cartoframes/utils/columns.py

```diff
@@ -100,115 +99,56 @@ def _extract_pgtype(fields):
 
 
 class DataframeColumnInfo(object):
-    def __init__(self, column, geom_column=None, geom_type=None, dtype=None):
-        if column:
-            self.dataframe = column
-            self.database = self._database_column_name(geom_column)
-            self.database_type = self._db_column_type(geom_column, geom_type, dtype)
+    def __init__(self, name, dtype=None, geom_type=None):
+        self.name = normalize_name(name)
+        if str(dtype) == 'geometry':
+            self.is_geom = True
+            self.type = 'geometry({}, 4326)'.format(geom_type or 'Point')
         else:
-            self.dataframe = None
-            self.database = Column.NORMALIZED_GEOM_COL_NAME
-            self.database_type = 'geometry(Point, 4326)'
+            self.is_geom = False
+            self.type = dtypes2pg(dtype)
 
     def __repr__(self):
-        return "DataframeColumnInfo<{}, {}, {}>".format(
-            self.dataframe,
-            self.database,
-            self.database_type
+        return "DataframeColumnInfo<{}, {}>".format(
+            self.name,
+            self.type
         )
 
-    def _database_column_name(self, geom_column):
-        if geom_column and self.dataframe == geom_column:
-            normalized_name = Column.NORMALIZED_GEOM_COL_NAME
-        else:
-            normalized_name = normalize_name(self.dataframe)
-
-        return normalized_name
-
-    def _db_column_type(self, geom_column, geom_type, dtype):
-        if geom_column and self.dataframe == geom_column:
-            db_type = 'geometry({}, 4326)'.format(geom_type or 'Point')
-        else:
-            db_type = dtypes2pg(dtype)
-
-        return db_type
-
-    def __eq__(self, obj):
-        if isinstance(obj, dict):
-            return self.dataframe == obj['dataframe'] and \
-                self.database == obj['database'] and \
-                self.database_type == obj['database_type']
-        else:
-            return self.dataframe == obj.dataframe and \
-                self.database == obj.database and \
-                self.database_type == obj.database_type
-
 
 class DataframeColumnsInfo(object):
-    def __init__(self, df, with_lnglat=None):
-        self.df = df
-        self.with_lnglat = with_lnglat
-
-        self.geom_column = self._get_geom_col_name()
-        geom_type, enc_type = self._get_geometry_type()
-        self.geom_type = geom_type
-        self.enc_type = enc_type
-
-        self.columns = self._get_columns_info()
+    def __init__(self, df):
```

Comment:
We are losing the relation between dataframe column names and the db column names. Is it a problem?

Reply:
No, because now they are the same.

Reply:
However, I'll take a second look at that part. Thanks!